src/python/pants/backend/python/subsystems/python_setup.py

```diff
@@ -26,7 +26,7 @@ class PythonSetup(Subsystem):
   @classmethod
   def register_options(cls, register):
     super(PythonSetup, cls).register_options(register)
-    register('--interpreter-constraints', advanced=True, default=['CPython>=2.7,<3'], type=list,
+    register('--interpreter-constraints', advanced=True, default=['CPython>=2.7,<3', 'CPython>=3.6,<3.8'], type=list,
```

Comment:
By choosing `<3.8` rather than `<4` we're changing from a policy of "We assume we'll be compatible with minor version updates unless we see otherwise" to the opposite. Is there something specific about 3.8 or 3.* that means we're doing this?